WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java

```diff
@@ -151,4 +153,47 @@ public void onCancel(DialogInterface dialog) {
         // Do not ask again
         AppPrefs.setImageOptimizePromoRequired(false);
     }
+
+    /**
+     * Given a media error returns the error message to display on the UI.
+     *
+     * @param error The media error occurred
+     * @return String  The associated error message.
+     */
+    public static String getErrorMessage(final Context context, final MediaModel media, final MediaStore.MediaError error) {
+        if (context == null || media == null || error == null) {
+            return null;
+        }
+
+        switch (error.type) {
+            case FS_READ_PERMISSION_DENIED:
+                return context.getString(R.string.error_media_insufficient_fs_permissions);
+            case NOT_FOUND:
+                return context.getString(R.string.error_media_not_found);
+            case AUTHORIZATION_REQUIRED:
+                return context.getString(R.string.media_error_no_permission_upload);
+            case REQUEST_TOO_LARGE:
+                if (media.isVideo()) {
+                    return context.getString(R.string.media_error_http_too_large_video_upload);
+                } else {
+                    return context.getString(R.string.media_error_http_too_large_photo_upload) + ". " +
+                            context.getString(R.string.media_error_suggest_optimize_image);
+                }
+            case SERVER_ERROR:
+                return context.getString(R.string.media_error_internal_server_error);
+            case TIMEOUT:
+                return context.getString(R.string.media_error_timeout);
+            case CONNECTION_ERROR:
+                return context.getString(R.string.connection_error) + ". "
+                        + context.getString(R.string.media_error_generic_connection_error);
+            case EXCEEDS_FILESIZE_LIMIT:
+                return context.getString(R.string.media_error_exceeds_php_filesize);
+            case EXCEEDS_MEMORY_LIMIT:
+                return context.getString(R.string.media_error_exceeds_memory_limit);
+            case PARSE_ERROR:
+                return context.getString(R.string.error_media_parse_error);
```

Comment:
"Could not parse response after media upload" is also quite cryptic. I'm assuming this means the server's response wasn't what we expected, and at this point we don't know whether the upload was successful? Or does it always mean the upload failed?